soc/arm/nxp_imx/rt/Kconfig.defconfig.series

```diff
@@ -159,14 +159,6 @@ choice SEGGER_SYSVIEW_SECTION
 	depends on SEGGER_SYSTEMVIEW
 endchoice
 
-#
-# MBEDTLS is larger but much faster than TinyCrypt so choose wisely
-#
-config MBEDTLS
-#config TINYCRYPT
-	default y if CSPRING_ENABLED
```

Comment:
Can't we just add the  `TINYCRYPT` or `MBEDTLS` dependencies for `CSPRING_ENABLED` option (didn't test it)? 
This way we can leave this option enabled by default.

<pre>
config CSPRING_ENABLED
#       bool "Cryptographically secure RNG functions enabled"
        bool
        default y
        depends on ENTROPY_HAS_DRIVER
        depends on ENTROPY_GENERATOR
	depends on MBEDTLS || TINYCRYPT
</pre>

Reply:
Yes, this looks like it might be better solution to how to manage CSPRING_ENABLED

Reply:
Unfortunately we cannot do that because of dependency loop.

CSPRING_ENABLED (defined at subsys/random/Kconfig:80), with definition...

config CSPRING_ENABLED
	bool
	default y
	depends on ENTROPY_HAS_DRIVER && ENTROPY_GENERATOR && (MBEDTLS || TINYCRYPT)

...depends on MBEDTLS (defined at modules/mbedtls/Kconfig:17), with definition...

menuconfig MBEDTLS
	bool "mbed TLS Support" if !MBEDTLS_PROMPTLESS
	depends on 0
	help
	  This option enables the mbedTLS cryptography library.

(select-related dependencies: (CRYPTO_MBEDTLS_SHIM && CRYPTO) || (WIFI_ESP32 && DT_HAS_ESPRESSIF_ESP32_WIFI_ENABLED && !SMP && WIFI) || (JWT_SIGN_RSA && CSPRING_ENABLED && ) || (UPDATEHUB_DTLS && UPDATEHUB) || (OSDP_SC_ENABLED && CSPRING_ENABLED && OSDP) || (OPENTHREAD_MBEDTLS && NET_L2_OPENTHREAD && NETWORKING) || (NET_TCP_ISN_RFC6528 && NET_TCP && !NET_RAW_MODE && NETWORKING) || (WEBSOCKET_CLIENT && NETWORKING) || (NET_SOCKETS_SOCKOPT_TLS && NET_NATIVE && NET_SOCKETS && NETWORKING) || (FLASH_AREA_CHECK_INTEGRITY_MBEDTLS && ))

...depends on the choice symbol JWT_SIGN_RSA (defined at subsys/jwt/Kconfig:17), with definition...

config JWT_SIGN_RSA
	bool "Use RSA signature (RS-2[56](https://github.com/zephyrproject-rtos/zephyr/actions/runs/4609113742/jobs/8145794711?pr=55377#step:7:57))"
	select MBEDTLS
	depends on CSPRING_ENABLED && 

...depends again on CSPRING_ENABLED (defined at subsys/random/Kconfig:80))
FAILED: CMakeFiles/html
I've replaced that with enabling CSPRING when MBEDTLS or TINYCRYPT is available.

Reply:
There is still dependency loop, I'm gonna take a look at it once again next week.

Reply:
@danieldegrasse @mmahadevan108 @dleach02 It seems I've fixed that but it required more work than I thought... There were a lot of dependency loops but I hope everything was tested under CI. Please take a look once again.